tools/sync/main.go

```diff
@@ -20,38 +20,51 @@ func main() {
 
 	ctx := context.Background()
 
+	sync := initLogging()
+	defer sync()
```

Comment:
@bobheadxi,  It is bit unclear / confusing to me 😅  on why the initialization of the logger being deferred to execute at the end ?

Reply:
@bobheadxi correct me if I'm wrong - the sync here is to make **sure** anything left on the background threads or still in memory buffers is flushed/persisted if or when the program exits. Think of it like a cleanup of some sorts

Reply:
@burmudar, yeah, the flush part makes sense to me why that is being deferred .
I guess confusion comes from trying to figure out where the sentry client is initialize... I was looking at the sentry doc example was trying to figure out where this step is happening :

```
err := sentry.Init(sentry.ClientOptions{
		Dsn: "https://examplePublicKey@o0.ingest.sentry.io/0",
	})
```
https://docs.sentry.io/platforms/go/

Is this where that sentry client is initiated: https://github.com/sourcegraph/clabot-config/pull/90/files#diff-5462a0d1d79a930e750a02cb1977089cb85ac9c80ff010b7457f87381632624bR12-R19 ?

Reply:
The Sentry client is initialized within the logging library :)

Reply:
It goes:

1. Add a log sink
2. Logging library takes sink configuration and does the necessary things to send error log entries to Sentry

Reply:
@bobheadxi  got it yeah that makes sense thank you 😄